spec/lib/appsignal/probes/gvl_spec.rb

```diff
@@ -4,6 +4,16 @@
 
   let(:hostname) { "some-host" }
 
+  around(:each) do |example|
+    real_program_name = $PROGRAM_NAME
+
+    begin
+      example.run
+    ensure
+      $PROGRAM_NAME = real_program_name
+    end
```

Comment:
This will work too without a new begin-end block:

```suggestion
    example.run
  ensure
    $PROGRAM_NAME = real_program_name
```